### PR TITLE
fix: store raw objname internally

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This patch fixes https://github.com/GreptimeTeam/greptimedb/issues/2595.

~~It's still a quick fix but not a hack. I'm going to go through other similar code path and trying to add a test if possible. But submit this PR for early review if any.~~

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/operator/src/statement.rs [hunk 1]
```diff
@@ -159,7 +159,11 @@ impl StatementExecutor {
                 self.truncate_table(table_name).await
             }
 
-            Statement::CreateDatabase(stmt) => {
+            Statement::CreateDatabase(mut stmt) => {
+                for ident in stmt.name.0.iter_mut() {
```
Comment (tisonkun) [inline src/operator/src/statement.rs:163]: Instead, we may write an utility to format `ObjectName`.

**Comment by waynexia** [comment]
Do you mind to add some [sqlness test](https://github.com/GreptimeTeam/greptimedb/blob/develop/tests/README.md) to cover this scenario? Thanks.

**Review by tisonkun — COMMENTED** [review]
Other names (table name) are handled by `table_idents_to_full_name` which omits quote style also.

**Comment by tisonkun** [comment]
> Do you mind to add some [sqlness test](https://github.com/GreptimeTeam/greptimedb/blob/develop/tests/README.md) to cover this scenario? Thanks.

@waynexia OK. How can I run a specify sqlness test case? `cargo sqlness` seems run all the tests.

File: tests/cases/standalone/common/create/create_database.result [hunk 2]
```diff
@@ -0,0 +1,19 @@
+create database illegal-database;
```
Comment (tisonkun) [inline tests/cases/standalone/common/create/create_database.result:1]: I just noticed that we don't support `drop database` yet? Shall we do that?

**Comment by waynexia** [comment]
> > Do you mind to add some [sqlness test](https://github.com/GreptimeTeam/greptimedb/blob/develop/tests/README.md) to cover this scenario? Thanks.
> 
> @waynexia OK. How can I run a specify sqlness test case? `cargo sqlness` seems run all the tests.

`cargo sqlness -t create_database` should do that for you. You can use `cargo sqlness --help` to see the manual.

File: tests/cases/standalone/common/create/create_database.result (hunk 2 above)
Comment (waynexia) [reply tests/cases/standalone/common/create/create_database.result:1]: This feature has not been implemented yet. Issue: https://github.com/GreptimeTeam/greptimedb/issues/825

**Comment by tisonkun** [comment]
@waynexia OK. You may give another review. This PR is ready IMO.

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**